modules/order/src/Form/UseMultiProfileConfirmForm.php

```diff
@@ -0,0 +1,390 @@
+          [
+            $order_ids,
+            $this->entity,
+          ],
+        ],
+      ],
+      'finished' => '\Drupal\commerce_order\MigrateExistingOrderProfiles::batchFinished',
+    ];
+
+    batch_set($batch);
+  }
+
+  /**
+   * Get the existing orders for this order type.
+   *
+   * @return array
+   *   Returns an array of order IDs.
+   */
+  protected function getExistingOrders() {
+    /** @var \Drupal\Core\Config\Entity\ConfigEntityType $bundle_entity_type */
+    $bundle_entity_type = $this->entityTypeManager->getDefinition($this->entity->getEntityTypeId());
+    /** @var \Drupal\Core\Entity\ContentEntityType $content_entity_type */
+    $content_entity_type = $this->entityTypeManager->getDefinition($bundle_entity_type->getBundleOf());
+    $orders = $this->entityTypeManager->getStorage($content_entity_type->id())
+      ->getQuery()
+      ->condition($content_entity_type->getKey('bundle'), $this->entity->id())
+      ->execute();
+
+    return $orders;
+  }
+
+  /**
+   * Returns a description on existing orders for this order type.
+   *
+   * @return string
+   *   The description text.
+   */
+  protected function getExistingOrderCountDescription() {
+    $description = '';
+
+    $orders = $this->getExistingOrders();
+
+    if ($order_count = count($orders)) {
+      $description = '<p>' . $this->formatPlural($order_count,
+          'The %type order type contains 1 order on your site.',
+          'The %type order type contains @count orders on your site.',
+          [
+            '%type' => $this->entity->label(),
+          ]) . '</p>';
+
+      $description .= '<p>' . $this->t('All of these existing orders will
+        be migrated to use the split shipping and billing profiles.'
+        ) . '</p>';
+    }
+
+    return $description;
+  }
+
+  /**
+   * Get all modules that will be affected and incompatible with the switch.
+   *
+   * @return array
+   *   An array of module names and the expected versions.
+   */
+  protected function getIncompatibleModules() {
+    $incompatible_modules = [];
+
+    $affected_modules = [
+      'commerce_pos' => '2.2',
+      'commerce_shipping' => '2.2',
+      'commerce_amws' => '2.2',
+    ];
+
+    foreach ($affected_modules as $module => $expected_version) {
+      $module_info = system_get_info('module', $module);
+      if (empty($module_info)) {
+        continue;
+      }
+
+      if (empty($module_info['version'])) {
+        continue;
+      }
+
+      // TODO: Find a reliable way to extract the module version.
+      $current_version = substr($module_info['version'], 4);
+
+      if (version_compare($current_version, $expected_version, '<')) {
+        $incompatible_modules[] = $this->t(
+          '@module_name at least version 8.x-@version or higher', [
+            '@module_name' => $module,
+            '@version' => $expected_version,
+          ]
+        );
+      }
+    }
+
+    return $incompatible_modules;
+  }
```

Comment:
**CHANGE REQUST**

That logic won't do I'm afraid. We don't know yet which module version will be supporting multiple profile types. Therefore, we need to be able to define either a version, or an empty string; an empty string will mean that the module is not yet at any version.